google/auth/jwt.py

```diff
@@ -426,13 +426,15 @@ def with_claims(self, issuer=None, subject=None, audience=None,
         Returns:
             google.auth.jwt.Credentials: A new credentials instance.
         """
+        new_additional_claims = self._additional_claims.copy()
+        new_additional_claims.update(additional_claims or {})
```

Comment:
If you made `additional_claims=()` by default you could ditch the `or {}` (FWIW `()` is not strictly needed, any "empty" iterable will do)